modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm

```diff
@@ -1657,3 +1657,183 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/sign/poster/contraband/korpstech, 32)
+	body_parts_covered = CHEST|ARMS // not really a buff, it has 0 armor inherently so this just ends up getting it shredded
+	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
+	/// The article of clothing we have absorbed and are emulating the effects of. Nullable.
+	var/obj/item/clothing/suit/absorbed_clothing
+	/// Any subtype of a typepath entered here will be insertable into the jacket.
+	var/static/list/obj/item/clothing/suit/clothing_we_can_absorb = list(
+		/obj/item/clothing/suit/toggle/labcoat,
+	)
+
+/obj/item/clothing/suit/costume/skyrat/vic_dresscoat_donator/Destroy()
+	drop_clothing()
+
+	return ..()
+
+
+/obj/item/clothing/suit/costume/skyrat/vic_dresscoat_donator/examine(mob/user)
+	. = ..()
+
+	if (!isnull(absorbed_clothing))
+		. += "It seems to have [absorbed_clothing] inside of it... you could try [span_boldnotice("using it")] to remove the clothing!"
+
+
+/obj/item/clothing/suit/costume/skyrat/vic_dresscoat_donator/attack_self(mob/user, modifiers)
+	if (isnull(absorbed_clothing) || !isliving(user))
+		return ..()
+
+	var/mob/living/living_user = user
+
+	user.balloon_alert_to_viewers("removing fitted clothing...")
+	if (!do_after(living_user, 3 SECONDS, src))
+		return
+	user.balloon_alert_to_viewers("clothing removed")
+	drop_clothing(user)
+
+
+/obj/item/clothing/suit/costume/skyrat/vic_dresscoat_donator/attackby(obj/item/attacking_item, mob/user, params)
+	if (iscarbon(user))
+		var/mob/living/carbon/carbon_user = user
+		if (carbon_user.combat_mode)
+			return ..()
+
+	for (var/obj/item/clothing/suit/absorbable_type as anything in clothing_we_can_absorb)
+		if (!istype(attacking_item, absorbable_type))
+			continue
+		absorb_clothing(attacking_item, user)
+		return FALSE
+
+	return ..()
+
+
+/// Consumes the article of clothing we are about to emulate, and applies its armor, storage, and suit storage variables to us.
+/obj/item/clothing/suit/costume/skyrat/vic_dresscoat_donator/proc/absorb_clothing(obj/item/clothing/suit/clothing_to_absorb, mob/living/user)
+	if (!isnull(absorbed_clothing))
+		balloon_alert(user, "already fitted!")
+		return FALSE
+	if (iscarbon(loc))
+		var/mob/living/carbon/carbon_loc = loc
+		if (src in carbon_loc.get_all_worn_items())
+			balloon_alert(user, "take it off first!")
+			return FALSE
+	if (!isnull(clothing_to_absorb.atom_storage))
+		var/atom/resolve_location = clothing_to_absorb.atom_storage.real_location?.resolve()
+		if (length(resolve_location?.contents))
+			clothing_to_absorb.balloon_alert(user, "take things out first!") // put it on the insertee so we go "oh THIS has stuff in it"
+			return FALSE
+
+	user.balloon_alert_to_viewers("inserting clothing...")
+	if (!do_after(user, 3 SECONDS, src))
+		return FALSE
+	user.balloon_alert_to_viewers("clothing inserted")
+	playsound(user, clothing_to_absorb.drop_sound, DROP_SOUND_VOLUME, FALSE)
+
+	absorbed_clothing = clothing_to_absorb
+	set_armor(clothing_to_absorb.get_armor())
```

Comment:
under no circumstances should donators be getting items that let them conceal armor in the visual of a non-armor clothing item

in fact, why is this not just a subtype of labcoat? what does this entire janky absorb thing serve other than letting this thing pretend to be a labcoat without being a labcoat?

Reply:
I originally made this with the intent of having it be able to mimic multiple types of inconsequential clothing, but quickly realized the labcoat was the only _truely_ inconsequential one I wanted to mimic

Winter coats' behavior are defined by their sprite, armor vests are a good old _fuck no_, and everything else doesn't really matter

Reply:
I COULD cut it all out and just subtype this onto labcoat, but 
1. The dresscoat simply is _not_ a labcoat, it's fabric is thin and basically only there to make a thin-ass dress
2. there may, in fact, come a day i realize there is another thing i want to wear under this

its a fair point though and i'll thinka bout it